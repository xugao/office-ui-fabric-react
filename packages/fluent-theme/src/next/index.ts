export { FluentTheme } from './FluentTheme';
export { getTokens } from './getTokens';
