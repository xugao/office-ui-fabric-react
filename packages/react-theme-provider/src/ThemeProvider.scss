/**
 * The ThemeProvider has a root consuming class which will apply the current
 * token values to the container DIV, so that fonts and default colors, which
 * refer to token values, can be applied.
 */
.body {
  background: var(--body-background);
  // color: var(--body-contentColor);
  // font-family: var(--body-fontFamily);
  // font-weight: var(--body-fontWeight);
  // font-size: var(--body-fontSize);
  // line-height: var(--body-fontLineHeight);

  // > * {
  //   box-sizing: border-box;
  // }
}
