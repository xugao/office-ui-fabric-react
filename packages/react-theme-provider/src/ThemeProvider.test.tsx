import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { mergeThemes } from '@fluentui/theme';
import { FluentTheme } from '@uifabric/fluent-theme/lib/next';
import { ThemeProvider } from './ThemeProvider';
import { Theme, PartialTheme } from './ThemeProvider.types';
import { useTheme } from './useTheme';
import { mount } from 'enzyme';

const lightTheme = mergeThemes({
  stylesheets: [],
  tokens: {
    body: {
      background: 'white',
      contentColor: 'black',
    },
  },
});

const darkTheme = mergeThemes({
  tokens: {
    body: {
      background: 'black',
      contentColor: 'white',
    },
  },
});

describe('ThemeProvider', () => {
  it('renders a div', () => {
    const component = renderer.create(<ThemeProvider>Hello</ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can handle a partial theme', () => {
    const partialTheme: PartialTheme = {
      tokens: {
        foo: {
          background: 'red',
        },
      },
    };

    const component = renderer.create(<ThemeProvider theme={partialTheme}>Hello</ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a div with styling', () => {
    const component = renderer.create(<ThemeProvider theme={lightTheme}>Hello</ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders nested themes', () => {
    const component = renderer.create(
      <ThemeProvider theme={lightTheme}>
        <div>Light theme</div>
        <ThemeProvider theme={darkTheme}>
          <div>Dark theme</div>
        </ThemeProvider>
      </ThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('provide the theme through context', () => {
    let resolvedTheme: Theme | undefined = undefined;
    const TestComponent = () => {
      resolvedTheme = useTheme();
      return null;
    };

    mount(
      <ThemeProvider theme={lightTheme}>
        <TestComponent />
      </ThemeProvider>,
    );

    const expectedTheme = mergeThemes(FluentTheme, lightTheme);
    expect(resolvedTheme).toEqual(expectedTheme);
  });
});
