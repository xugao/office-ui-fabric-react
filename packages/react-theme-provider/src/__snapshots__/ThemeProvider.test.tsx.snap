// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider can apply body theme to body 1`] = `
<body
  class=

      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: black;
        --color-body-contentColor: white;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checked-iconColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-checkedHovered-iconColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: #f3f2f1;
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: #a19f9d;
        --color-brand-disabled-secondaryContentColor: #a19f9d;
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: #0078d4;
        --color-brand-focused-borderColor: transparent;
        --color-brand-focused-contentColor: #ffffff;
        --color-brand-focused-iconColor: #ffffff;
        --color-brand-focused-secondaryContentColor: #ffffff;
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: transparent;
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: transparent;
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: #ffffff;
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
        --text-variant-body-fontSize: 14px;
        --text-variant-body-fontWeight: 400;
        --text-variant-body-lineHeight: 20px;
        --text-variant-caption-fontSize: 12px;
        --text-variant-caption-fontWeight: 400;
        --text-variant-caption-lineHeight: 14px;
        --text-variant-display-fontSize: 68px;
        --text-variant-display-fontWeight: 600;
        --text-variant-display-lineHeight: 92px;
        --text-variant-headline-fontSize: 20px;
        --text-variant-headline-fontWeight: 600;
        --text-variant-headline-lineHeight: 28px;
        --text-variant-largeTitle-fontSize: 40px;
        --text-variant-largeTitle-fontWeight: 600;
        --text-variant-largeTitle-lineHeight: 52px;
        --text-variant-subHeadline-fontSize: 16px;
        --text-variant-subHeadline-fontWeight: 600;
        --text-variant-subHeadline-lineHeight: 22px;
        --text-variant-title1-fontSize: 32px;
        --text-variant-title1-fontWeight: 600;
        --text-variant-title1-lineHeight: 40px;
        --text-variant-title2-fontSize: 28px;
        --text-variant-title2-fontWeight: 600;
        --text-variant-title2-lineHeight: 36px;
        --text-variant-title3-fontSize: 24px;
        --text-variant-title3-fontWeight: 600;
        --text-variant-title3-lineHeight: 32px;
      }
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: black;
        color: white;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
/>
`;

exports[`ThemeProvider can apply body theme to body 2`] = `
<div
  className=
      foo
      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: black;
        --color-body-contentColor: white;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checked-iconColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-checkedHovered-iconColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: #f3f2f1;
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: #a19f9d;
        --color-brand-disabled-secondaryContentColor: #a19f9d;
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: #0078d4;
        --color-brand-focused-borderColor: transparent;
        --color-brand-focused-contentColor: #ffffff;
        --color-brand-focused-iconColor: #ffffff;
        --color-brand-focused-secondaryContentColor: #ffffff;
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: transparent;
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: transparent;
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: #ffffff;
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
        --text-variant-body-fontSize: 14px;
        --text-variant-body-fontWeight: 400;
        --text-variant-body-lineHeight: 20px;
        --text-variant-caption-fontSize: 12px;
        --text-variant-caption-fontWeight: 400;
        --text-variant-caption-lineHeight: 14px;
        --text-variant-display-fontSize: 68px;
        --text-variant-display-fontWeight: 600;
        --text-variant-display-lineHeight: 92px;
        --text-variant-headline-fontSize: 20px;
        --text-variant-headline-fontWeight: 600;
        --text-variant-headline-lineHeight: 28px;
        --text-variant-largeTitle-fontSize: 40px;
        --text-variant-largeTitle-fontWeight: 600;
        --text-variant-largeTitle-lineHeight: 52px;
        --text-variant-subHeadline-fontSize: 16px;
        --text-variant-subHeadline-fontWeight: 600;
        --text-variant-subHeadline-lineHeight: 22px;
        --text-variant-title1-fontSize: 32px;
        --text-variant-title1-fontWeight: 600;
        --text-variant-title1-lineHeight: 40px;
        --text-variant-title2-fontSize: 28px;
        --text-variant-title2-fontWeight: 600;
        --text-variant-title2-lineHeight: 36px;
        --text-variant-title3-fontSize: 24px;
        --text-variant-title3-fontWeight: 600;
        --text-variant-title3-lineHeight: 32px;
      }
>
  app
</div>
`;

exports[`ThemeProvider can apply body theme to none 1`] = `
<div
  className=
      foo
      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: black;
        --color-body-contentColor: white;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checked-iconColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-checkedHovered-iconColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: #f3f2f1;
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: #a19f9d;
        --color-brand-disabled-secondaryContentColor: #a19f9d;
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: #0078d4;
        --color-brand-focused-borderColor: transparent;
        --color-brand-focused-contentColor: #ffffff;
        --color-brand-focused-iconColor: #ffffff;
        --color-brand-focused-secondaryContentColor: #ffffff;
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: transparent;
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: transparent;
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: #ffffff;
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
        --text-variant-body-fontSize: 14px;
        --text-variant-body-fontWeight: 400;
        --text-variant-body-lineHeight: 20px;
        --text-variant-caption-fontSize: 12px;
        --text-variant-caption-fontWeight: 400;
        --text-variant-caption-lineHeight: 14px;
        --text-variant-display-fontSize: 68px;
        --text-variant-display-fontWeight: 600;
        --text-variant-display-lineHeight: 92px;
        --text-variant-headline-fontSize: 20px;
        --text-variant-headline-fontWeight: 600;
        --text-variant-headline-lineHeight: 28px;
        --text-variant-largeTitle-fontSize: 40px;
        --text-variant-largeTitle-fontWeight: 600;
        --text-variant-largeTitle-lineHeight: 52px;
        --text-variant-subHeadline-fontSize: 16px;
        --text-variant-subHeadline-fontWeight: 600;
        --text-variant-subHeadline-lineHeight: 22px;
        --text-variant-title1-fontSize: 32px;
        --text-variant-title1-fontWeight: 600;
        --text-variant-title1-lineHeight: 40px;
        --text-variant-title2-fontSize: 28px;
        --text-variant-title2-fontWeight: 600;
        --text-variant-title2-lineHeight: 36px;
        --text-variant-title3-fontSize: 24px;
        --text-variant-title3-fontWeight: 600;
        --text-variant-title3-lineHeight: 32px;
      }
>
  app
</div>
`;

exports[`ThemeProvider can handle a partial theme 1`] = `
<div
  className=

      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: #ffffff;
        --color-body-contentColor: #323130;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checked-iconColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-checkedHovered-iconColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: #f3f2f1;
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: #a19f9d;
        --color-brand-disabled-secondaryContentColor: #a19f9d;
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: #0078d4;
        --color-brand-focused-borderColor: transparent;
        --color-brand-focused-contentColor: #ffffff;
        --color-brand-focused-iconColor: #ffffff;
        --color-brand-focused-secondaryContentColor: #ffffff;
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: transparent;
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: transparent;
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: #ffffff;
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
        --foo-background: red;
        --text-variant-body-fontSize: 14px;
        --text-variant-body-fontWeight: 400;
        --text-variant-body-lineHeight: 20px;
        --text-variant-caption-fontSize: 12px;
        --text-variant-caption-fontWeight: 400;
        --text-variant-caption-lineHeight: 14px;
        --text-variant-display-fontSize: 68px;
        --text-variant-display-fontWeight: 600;
        --text-variant-display-lineHeight: 92px;
        --text-variant-headline-fontSize: 20px;
        --text-variant-headline-fontWeight: 600;
        --text-variant-headline-lineHeight: 28px;
        --text-variant-largeTitle-fontSize: 40px;
        --text-variant-largeTitle-fontWeight: 600;
        --text-variant-largeTitle-lineHeight: 52px;
        --text-variant-subHeadline-fontSize: 16px;
        --text-variant-subHeadline-fontWeight: 600;
        --text-variant-subHeadline-lineHeight: 22px;
        --text-variant-title1-fontSize: 32px;
        --text-variant-title1-fontWeight: 600;
        --text-variant-title1-lineHeight: 40px;
        --text-variant-title2-fontSize: 28px;
        --text-variant-title2-fontWeight: 600;
        --text-variant-title2-lineHeight: 36px;
        --text-variant-title3-fontSize: 24px;
        --text-variant-title3-fontWeight: 600;
        --text-variant-title3-lineHeight: 32px;
      }
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  Hello
</div>
`;

exports[`ThemeProvider renders a div 1`] = `
<div
  className=

      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: #ffffff;
        --color-body-contentColor: #323130;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checked-iconColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-checkedHovered-iconColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: #f3f2f1;
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: #a19f9d;
        --color-brand-disabled-secondaryContentColor: #a19f9d;
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: #0078d4;
        --color-brand-focused-borderColor: transparent;
        --color-brand-focused-contentColor: #ffffff;
        --color-brand-focused-iconColor: #ffffff;
        --color-brand-focused-secondaryContentColor: #ffffff;
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: transparent;
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: transparent;
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: #ffffff;
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
        --text-variant-body-fontSize: 14px;
        --text-variant-body-fontWeight: 400;
        --text-variant-body-lineHeight: 20px;
        --text-variant-caption-fontSize: 12px;
        --text-variant-caption-fontWeight: 400;
        --text-variant-caption-lineHeight: 14px;
        --text-variant-display-fontSize: 68px;
        --text-variant-display-fontWeight: 600;
        --text-variant-display-lineHeight: 92px;
        --text-variant-headline-fontSize: 20px;
        --text-variant-headline-fontWeight: 600;
        --text-variant-headline-lineHeight: 28px;
        --text-variant-largeTitle-fontSize: 40px;
        --text-variant-largeTitle-fontWeight: 600;
        --text-variant-largeTitle-lineHeight: 52px;
        --text-variant-subHeadline-fontSize: 16px;
        --text-variant-subHeadline-fontWeight: 600;
        --text-variant-subHeadline-lineHeight: 22px;
        --text-variant-title1-fontSize: 32px;
        --text-variant-title1-fontWeight: 600;
        --text-variant-title1-lineHeight: 40px;
        --text-variant-title2-fontSize: 28px;
        --text-variant-title2-fontWeight: 600;
        --text-variant-title2-lineHeight: 36px;
        --text-variant-title3-fontSize: 24px;
        --text-variant-title3-fontWeight: 600;
        --text-variant-title3-lineHeight: 32px;
      }
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  Hello
</div>
`;

exports[`ThemeProvider renders a div with styling 1`] = `
<div
  className=

      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: white;
        --color-body-contentColor: black;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checked-iconColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-checkedHovered-iconColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: #f3f2f1;
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: #a19f9d;
        --color-brand-disabled-secondaryContentColor: #a19f9d;
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: #0078d4;
        --color-brand-focused-borderColor: transparent;
        --color-brand-focused-contentColor: #ffffff;
        --color-brand-focused-iconColor: #ffffff;
        --color-brand-focused-secondaryContentColor: #ffffff;
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: transparent;
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: transparent;
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: #ffffff;
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
        --text-variant-body-fontSize: 14px;
        --text-variant-body-fontWeight: 400;
        --text-variant-body-lineHeight: 20px;
        --text-variant-caption-fontSize: 12px;
        --text-variant-caption-fontWeight: 400;
        --text-variant-caption-lineHeight: 14px;
        --text-variant-display-fontSize: 68px;
        --text-variant-display-fontWeight: 600;
        --text-variant-display-lineHeight: 92px;
        --text-variant-headline-fontSize: 20px;
        --text-variant-headline-fontWeight: 600;
        --text-variant-headline-lineHeight: 28px;
        --text-variant-largeTitle-fontSize: 40px;
        --text-variant-largeTitle-fontWeight: 600;
        --text-variant-largeTitle-lineHeight: 52px;
        --text-variant-subHeadline-fontSize: 16px;
        --text-variant-subHeadline-fontWeight: 600;
        --text-variant-subHeadline-lineHeight: 22px;
        --text-variant-title1-fontSize: 32px;
        --text-variant-title1-fontWeight: 600;
        --text-variant-title1-lineHeight: 40px;
        --text-variant-title2-fontSize: 28px;
        --text-variant-title2-fontWeight: 600;
        --text-variant-title2-lineHeight: 36px;
        --text-variant-title3-fontSize: 24px;
        --text-variant-title3-fontWeight: 600;
        --text-variant-title3-lineHeight: 32px;
      }
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: white;
        color: black;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  Hello
</div>
`;

exports[`ThemeProvider renders nested themes 1`] = `
<div
  className=

      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: white;
        --color-body-contentColor: black;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checked-iconColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-checkedHovered-iconColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: #f3f2f1;
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: #a19f9d;
        --color-brand-disabled-secondaryContentColor: #a19f9d;
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: #0078d4;
        --color-brand-focused-borderColor: transparent;
        --color-brand-focused-contentColor: #ffffff;
        --color-brand-focused-iconColor: #ffffff;
        --color-brand-focused-secondaryContentColor: #ffffff;
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: transparent;
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: transparent;
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: #ffffff;
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
        --text-variant-body-fontSize: 14px;
        --text-variant-body-fontWeight: 400;
        --text-variant-body-lineHeight: 20px;
        --text-variant-caption-fontSize: 12px;
        --text-variant-caption-fontWeight: 400;
        --text-variant-caption-lineHeight: 14px;
        --text-variant-display-fontSize: 68px;
        --text-variant-display-fontWeight: 600;
        --text-variant-display-lineHeight: 92px;
        --text-variant-headline-fontSize: 20px;
        --text-variant-headline-fontWeight: 600;
        --text-variant-headline-lineHeight: 28px;
        --text-variant-largeTitle-fontSize: 40px;
        --text-variant-largeTitle-fontWeight: 600;
        --text-variant-largeTitle-lineHeight: 52px;
        --text-variant-subHeadline-fontSize: 16px;
        --text-variant-subHeadline-fontWeight: 600;
        --text-variant-subHeadline-lineHeight: 22px;
        --text-variant-title1-fontSize: 32px;
        --text-variant-title1-fontWeight: 600;
        --text-variant-title1-lineHeight: 40px;
        --text-variant-title2-fontSize: 28px;
        --text-variant-title2-fontWeight: 600;
        --text-variant-title2-lineHeight: 36px;
        --text-variant-title3-fontSize: 24px;
        --text-variant-title3-fontWeight: 600;
        --text-variant-title3-lineHeight: 32px;
      }
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: white;
        color: black;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  <div>
    Light theme
  </div>
  <div
    className=

        {
          --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          --body-fontSize: 14px;
          --body-fontWeight: 400;
          --body-mozOsxFontSmoothing: grayscale;
          --body-webkitFontSmoothing: antialiased;
          --color-body-background: black;
          --color-body-contentColor: white;
          --color-brand-background: #0078d4;
          --color-brand-borderColor: transparent;
          --color-brand-checked-background: #005a9e;
          --color-brand-checked-contentColor: #ffffff;
          --color-brand-checked-iconColor: #ffffff;
          --color-brand-checkedHovered-background: #005a9e;
          --color-brand-checkedHovered-contentColor: #ffffff;
          --color-brand-checkedHovered-iconColor: #ffffff;
          --color-brand-contentColor: #ffffff;
          --color-brand-disabled-background: #f3f2f1;
          --color-brand-disabled-borderColor: #f3f2f1;
          --color-brand-disabled-contentColor: #a19f9d;
          --color-brand-disabled-dividerColor: #c8c6c4;
          --color-brand-disabled-iconColor: #a19f9d;
          --color-brand-disabled-secondaryContentColor: #a19f9d;
          --color-brand-dividerColor: #ffffff;
          --color-brand-focused-background: #0078d4;
          --color-brand-focused-borderColor: transparent;
          --color-brand-focused-contentColor: #ffffff;
          --color-brand-focused-iconColor: #ffffff;
          --color-brand-focused-secondaryContentColor: #ffffff;
          --color-brand-hovered-background: #106ebe;
          --color-brand-hovered-borderColor: transparent;
          --color-brand-hovered-contentColor: #ffffff;
          --color-brand-hovered-secondaryContentColor: #ffffff;
          --color-brand-iconColor: #ffffff;
          --color-brand-pressed-background: #005a9e;
          --color-brand-pressed-borderColor: transparent;
          --color-brand-pressed-contentColor: #ffffff;
          --color-brand-pressed-iconColor: #ffffff;
          --color-brand-pressed-secondaryContentColor: #ffffff;
          --color-brand-secondaryContentColor: #ffffff;
          --text-variant-body-fontSize: 14px;
          --text-variant-body-fontWeight: 400;
          --text-variant-body-lineHeight: 20px;
          --text-variant-caption-fontSize: 12px;
          --text-variant-caption-fontWeight: 400;
          --text-variant-caption-lineHeight: 14px;
          --text-variant-display-fontSize: 68px;
          --text-variant-display-fontWeight: 600;
          --text-variant-display-lineHeight: 92px;
          --text-variant-headline-fontSize: 20px;
          --text-variant-headline-fontWeight: 600;
          --text-variant-headline-lineHeight: 28px;
          --text-variant-largeTitle-fontSize: 40px;
          --text-variant-largeTitle-fontWeight: 600;
          --text-variant-largeTitle-lineHeight: 52px;
          --text-variant-subHeadline-fontSize: 16px;
          --text-variant-subHeadline-fontWeight: 600;
          --text-variant-subHeadline-lineHeight: 22px;
          --text-variant-title1-fontSize: 32px;
          --text-variant-title1-fontWeight: 600;
          --text-variant-title1-lineHeight: 40px;
          --text-variant-title2-fontSize: 28px;
          --text-variant-title2-fontWeight: 600;
          --text-variant-title2-lineHeight: 36px;
          --text-variant-title3-fontSize: 24px;
          --text-variant-title3-fontWeight: 600;
          --text-variant-title3-lineHeight: 32px;
        }
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          background: black;
          color: white;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
        }
  >
    <div>
      Dark theme
    </div>
  </div>
</div>
`;
